spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/db/migration/AbstractMigrateUriRegistrySqlCommand.java

```diff
@@ -0,0 +1,211 @@
+	 * required to insert data based on list of
+	 * {@code AppRegistrationMigrationData}'s and keep {@code hibernate_sequence}
+	 * up-to-date.
+	 *
+	 * @param jdbcTemplate the jdbc template
+	 * @param data the data
+	 */
+	protected abstract void updateAppRegistration(JdbcTemplate jdbcTemplate, List<AppRegistrationMigrationData> data);
+
+	/**
+	 * Creates a migration data from URI_REGISTRY table to get inserted
+	 * into app_registration table. We're working on a raw data level
+	 * thus no use hibernate, spring repositories nor entity classes.
+	 *
+	 * @param jdbcTemplate the jdbc template
+	 * @return the list
+	 */
+	protected List<AppRegistrationMigrationData> createAppRegistrationMigrationData(JdbcTemplate jdbcTemplate) {
+		Map<String, AppRegistrationMigrationData> data = new HashMap<>();
+		// to get version using same existing logic
+		AppResourceCommon arc = new AppResourceCommon(new MavenProperties(), new DelegatingResourceLoader());
+		// track that we only add default version for one by just giving it to first one
+		// just to be on a safe side
+		Collection<String> defaultVersions = new HashSet<>();
+
+		// format for old URI_REGISTRY is:
+		// +-------------------------------+-------------------------------------------------------------------------------------------------+
+		// | NAME                          | URI                                                                                             |
+		// +-------------------------------+-------------------------------------------------------------------------------------------------+
+		// | processor.bridge              | maven://org.springframework.cloud.stream.app:bridge-processor-rabbit:2.0.1.RELEASE              |
+		// | processor.bridge.metadata     | maven://org.springframework.cloud.stream.app:bridge-processor-rabbit:jar:metadata:2.0.1.RELEASE |
+		//
+		// this needs to go to a new app_registration table structure where
+		// uri and metadata and other fields are within same record.
+		// AppRegistrationMigrationData is base of that info which then
+		// gets actually updates with updateAppRegistration method.
+
+		jdbcTemplate.query("select NAME, URI from URI_REGISTRY", rs -> {
+			AppRegistrationMigrationData armd;
+			String name = rs.getString(1);
+			String uri = rs.getString(2);
+			// we assume i.e processor.bridge and processor.bridge.metadata belong together
+			String[] split = name.split("\\.");
+			if (split.length == 2 || split.length == 3) {
+				String key = split[0] + split[1];
+				armd = data.getOrDefault(key, new AppRegistrationMigrationData());
+				if (!defaultVersions.contains(key)) {
+					armd.setDefaultVersion(true);
+					defaultVersions.add(key);
+				}
+				if (split.length == 2) {
+					// we got *.* first
+					armd.setName(split[1]);
+					armd.setType(ApplicationType.valueOf(split[0]).ordinal());
+					armd.setUri(uri);
+					armd.setVersion(arc.getResourceVersion(arc.getResource(uri)));
```

Comment:
I guess we need to discuss how to inform users who are in classic mode and don't have a version field that their stream def isn't going to be migrated.  Also, I believe the old table gets dropped, so we prob shouldn't do that if we could not migrate a registered app....